test/topology_custom/test_no_removed_node_event_on_ip_change.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Test clusters can restart fine after an IP address change.
+"""
+
+import logging
+import time
+
+import pytest
+from cassandra.cluster import Cluster, Session
+from cassandra.policies import WhiteListRoundRobinPolicy
+from cassandra.pool import Host  # type: ignore # pylint: disable=no-name-in-module
+from test.pylib.manager_client import ManagerClient
+from test.pylib.internal_types import ServerInfo
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.conftest import cluster_con
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_no_removed_node_event_on_ip_change(manager: ManagerClient, caplog):
+    logger.info("starting the first node (leader)")
+    servers = [await manager.server_add()]
+
+    logger.info("starting the second node (follower)")
+    servers += [await manager.server_add()]
+
+    logger.info("close the default manager driver")
+    manager.driver_close()
+
+    logger.info(f"stopping the follower node {servers[1]}")
+    await manager.server_stop_gracefully(servers[1].server_id)
+
+    s1_old_ip = servers[1].ip_addr
+    s1_new_ip = await manager.server_change_ip(servers[1].server_id)
+    logger.info(f"changed {servers[1]} ip to {s1_new_ip}")
+
+    logger.info(f"starting the follower node {servers[1]}")
+    await manager.server_start(servers[1].server_id)
+    servers[1] = ServerInfo(servers[1].server_id, s1_new_ip, s1_new_ip)
+
+    logger.info("creating driver")
+    # We use load_balancing_policy to ensure that the control connection
+    # is connected to the first node, so that we get the TOPOLOGY_CHANGE notifications.
+    test_cluster: Cluster = cluster_con([servers[0].ip_addr, servers[1].ip_addr], 9042, False,
```

Comment:
Should use `with cluster_con(...)` and `with test_cluster.connect()` to not leak resources, although it probably doesn't matter since IIUC every test file is run in its own process

Reply:
ok, will fix